data_structures/stack.hpp

```diff
@@ -1,10 +1,10 @@
 /**
- * @file stack.h
+ * @file stack.hpp
```

Comment:
No need to add the filename. It's best to let Doxygen automatically detect it, otherwise, you'd have to specify the whole directory as well.
```suggestion
 * @file
```